src/main/java/hudson/plugins/gradle/BuildScanLogScanner.java

```diff
@@ -1,31 +1,37 @@
 package hudson.plugins.gradle;
 
+import java.util.Optional;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
 public class BuildScanLogScanner {
+    private static final Pattern BUILD_SCAN_PATTERN = Pattern.compile("Publishing (build scan|build information)\\.\\.\\.");
+    private static final Pattern URL_PATTERN = Pattern.compile(".*(?:\\[INFO] )?(https?://.*/s/.*)");
+    private static final int LINES_TO_SCAN = 1000;
     private final BuildScanPublishedListener listener;
     private int linesSinceBuildScanPublishingMessage = Integer.MAX_VALUE;
-    private static final Pattern BUILD_SCAN_PATTERN = Pattern.compile("Publishing (build scan|build information)\\.\\.\\.");
-    private static final Pattern URL_PATTERN = Pattern.compile("https?://\\S*");
 
     public BuildScanLogScanner(BuildScanPublishedListener listener) {
         this.listener = listener;
     }
 
     void scanLine(String line) {
-        if (linesSinceBuildScanPublishingMessage < 10) {
+        if (linesSinceBuildScanPublishingMessage < LINES_TO_SCAN) {
             linesSinceBuildScanPublishingMessage++;
-            Matcher matcher = URL_PATTERN.matcher(line);
-            if (matcher.find()) {
+            Optional<String> maybeBuildScanUrl = doGetBuildScanId(line);
+            if (maybeBuildScanUrl.isPresent()) {
                 linesSinceBuildScanPublishingMessage = Integer.MAX_VALUE;
-                String buildScanUrl = matcher.group();
-                listener.onBuildScanPublished(buildScanUrl);
+                listener.onBuildScanPublished(maybeBuildScanUrl.get());
             }
         }
         if (BUILD_SCAN_PATTERN.matcher(line).find()) {
             linesSinceBuildScanPublishingMessage = 0;
         }
 
     }
+
+    private static Optional<String> doGetBuildScanId(String text) {
```

Comment:
Maybe call it `tryFindBuildScanId(...)`, so that the name better reflects that the output might be `Optional`. WDYT?

Reply:
`tryFindBuildScanUrl` since it's the url and not the id that it extracts.